api/mutations/thread/publishThread.js

```diff
@@ -52,33 +67,21 @@ export default async (
     );
   }
 
-  // get the current user's permissions in the channel where the thread is being posted
-  const getCurrentUserChannelPermissions = getUserPermissionsInChannel(
-    thread.channelId,
-    currentUser.id
-  );
-
-  const getCurrentUserCommunityPermissions = getUserPermissionsInCommunity(
-    thread.communityId,
-    currentUser.id
-  );
-
-  const getChannel = getChannelById(thread.channelId);
-  const getCommunity = getCommunityById(thread.communityId);
-
   const [
     currentUserChannelPermissions,
     currentUserCommunityPermissions,
     channel,
     community,
+    usersPreviousPublishedThreads,
   ] = await Promise.all([
-    getCurrentUserChannelPermissions,
-    getCurrentUserCommunityPermissions,
-    getChannel,
-    getCommunity,
+    getUserPermissionsInChannel(thread.channelId, currentUser.id),
+    getUserPermissionsInCommunity(thread.communityId, currentUser.id),
+    getChannelById(thread.channelId),
+    getCommunityById(thread.communityId),
+    getThreadsByUserAsSpamCheck(currentUser.id),
```

Comment:
These should all be loaders. We already have the first 4, only the new one doesn't have a loader.

Reply:
Not totally sure why we'd want a loader for the last one, or even the value of loaders in mutations (they have way more probability of causing cache mismatches than of providing meaningful mutation speed, imo). What do you think?